examples/trusted_database/module_0/rust/src/lib.rs

```diff
@@ -0,0 +1,95 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//! Trusted Database example for Project Oak.
+//!
+//! This shows how an Oak Node can host an in-memory database and process private queries from
+//! clients.
+//!
+//! Current example implementation uses a list of Santander Cycles in London as test database:
+//! https://tfl.gov.uk/tfl/syndication/feeds/cycle-hire/livecyclehireupdates.xml
+//!
+//! Clients send their location coordinates (latitude and longitude) and the Oak Node returns the
+//! location of the closest Point Of Interest (in our case a bike station).
+
+mod database;
+#[cfg(test)]
+mod tests;
+
+use database::load_database;
+use log::debug;
+use oak::{grpc, io::SenderExt, proto::oak::invocation::GrpcInvocationReceiver, Node};
+use trusted_database_handler::proto::oak::examples::trusted_database::{
+    PointOfInterest, TrustedDatabaseCommand,
+};
+
+/// Oak Node that contains an in-memory database.
+pub struct TrustedDatabaseNode {
+    points_of_interest: Vec<PointOfInterest>,
+}
+
+impl Node<grpc::Invocation> for TrustedDatabaseNode {
+    fn handle_command(&mut self, invocation: grpc::Invocation) -> Result<(), oak::OakError> {
+        // Create a client request handler Node.
+        debug!("Creating handler Node");
+        let (sender, receiver) = oak::io::channel_create::<TrustedDatabaseCommand>()
+            .expect("Couldn't create command channel");
+        oak::node_create(
+            &oak::node_config::wasm("trusted_database_handler", "handler_oak_main"),
+            receiver.handle,
+        )
```

Comment:
Am I missing something, or is this not actually doing anything with respect to labels?

Reply:
I think since `module_0` only proxies invocation - it does not interact with labels

Reply:
Then it's not doing much in terms of Oak specific functionality, right? Or in other words, when we add labels from the client point of view, this logic will have to change to propagate them correctly, which could have been done now, but I guess we can postpone. But do you agree that if it doesn't do anything label related, then it's probably wrong? Or is the magic happening elsewhere?

Reply:
I don't think is possible to create a node with the right label until #1406 is resolved. Perhaps add a TODO to add the label logic here once #1406 is complete, and as part of adding more restrictive labels to the client. When the client label is more restrictive, the trusted_database_handler will not be able to read the invocation unless it has the same label.

Reply:
You are right @conradgrobler , thanks!